pkg/scheduler/service/cleaner.go

```diff
@@ -61,3 +125,99 @@ func (c *cleaner) purgeReconciliations(transition *ClusterStatusTransition, conf
 		}
 	}
 }
+
+func (c *cleaner) getLatestReconciliations(latestEntitiesCount int, transition *ClusterStatusTransition) ([]*model.ReconciliationEntity, error) {
+	return transition.ReconciliationRepository().GetReconciliations(&reconciliation.Limit{Count: latestEntitiesCount})
+}
+
+func (c *cleaner) dropRecordsOlderThan(t time.Time, transition *ClusterStatusTransition) error {
```

Comment:
I think we can extract the shared logic of `dropRecordsOlderThan` and `dropSuccessfulRecordsOlderThan` to something like `dropRecordsOlderThanByFilter(t time.Time, transition *ClusterStatusTransition,  filter func(*model.ReconciliationEntity) bool)` to get rid of duplicate code

Reply:
Done